### DISCUSSION TITLE
Errors following documentation for Modeling Entity Relationships v5.9

### DISCUSSION THREAD

**Discussion by bjss-jordan** [mikro-orm/mikro-orm#4922] [General]
Hello,

I am using version 5.9.3 and I get compilation errors when following the examples for one-to-many and many-to-many here: https://mikro-orm.io/docs/relationships

For example:

```
  @OneToMany(() => Version, (version) => version.product)
  versions = new Collection<Version>(this)
```

gives me the error: 

> Argument type () => Version is not assignable to parameter type string | ((e?: any) => EntityName<Version>) 

Have I somehow ended up with the wrong version or are the docs wrong? Or is it tsconfig settings or something else?

many thanks!

**Comment by B4nan** [comment]
Hard to say without seeing some details, the docs are correct, nothing changed about this, there are working tests as well as many example apps.

A similar (or maybe the same) error happens if you upgrade to v6 and forget to remove the generic type args from the ORM base entity, but since you said you are on v5, I don't think its relevant.

**Reply by bjss-jordan** [reply]
OK thanks. Something strange is going on because Entity Generator generated a some invalid stuff too.

These are my package versions:

  "dependencies": {
    "@mikro-orm/cli": "^5.9.3",
    "@mikro-orm/core": "^5.9.3",
    "@mikro-orm/entity-generator": "^5.9.3",
    "@mikro-orm/nestjs": "^5.2.2",
    "@mikro-orm/postgresql": "^5.9.3",

I'll download an example app and see if I can spot any differences.

**Reply by B4nan** [reply]
ideally provide a complete repro, there are many things that can matter, e.g. tsconfig, ESM/CJS, TS version.

FYI entity generator is there for easy bootstrap on existing schema, its not a schema-first approach (you are not supposed to be calling that repetitively). it might not generate 100% correct output (although in most cases it should), you are always supposed to be checking and fixing that manually. what invalid stuff it generates?

**Reply by bjss-jordan** [reply]
It seems to be tsconfig (I swapped in the tsconfig for one of the example apps and the errors went away)

This is my tsconfig:

```
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "target": "ES2021",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "skipLibCheck": true,
    "strictNullChecks": true,
    "noImplicitAny": true,
    "strictBindCallApply": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
  }
}
```

**Reply by B4nan** [reply]
looks like you are missing `esModuleInterop` option (and you can drop `allowSyntheticDefaultImports` as thats implied from it iirc)

**Reply by bjss-jordan** [reply]
Great, thankyou! Much appreciated

**Reply by B4nan** [reply]
i keep considering whether i should disable it, but then the code can be a bit messy, i think its generally good to use this flag, especially since using modern setup will imply it anyway (with `module: 'node16'` or `nodenext` you get it set to true automatically)

**Reply by bjss-jordan** [reply]
Really sorry I was wrong, I've come back to this today and the error is back.

However, I've now realised that 'npm run build' works fine, but Intellij Ultimate is still displaying the error.

So I guess something is wrong with my IDE setup.
